src/functions/module_statistical.ts

```diff
@@ -149,6 +135,228 @@ function centile(
   return percentile(sortedArray, _percent, isInclusive);
 }
 
+function prepareDataForRegression(X: Matrix<number>, Y: Matrix<number>, newX: Matrix<number>) {
+  const _X = X.length ? X : [range(1, Y.flat().length + 1)];
+  const n_var = _X.length;
+  let _newX = newX.length ? newX : _X;
+  _newX = _newX.length === n_var ? transpose2dArray(_newX) : _newX;
+  return { _X, _newX };
+}
+
+/*
+ * This function performs a linear regression on the data set. It returns an array with two elements.
+ * The first element is the slope, and the second element is the intercept.
+ * The linear regression line is: y = slope*x + intercept
+ * The function use the least squares method to find the best fit for the data set :
+ * see https://www.mathsisfun.com/data/least-squares-regression.html
+ *     https://www.statology.org/standard-error-of-estimate/
+ *     https://agronomy4future.org/?p=16670
+ *     https://vitalflux.com/interpreting-f-statistics-in-linear-regression-formula-examples/
+ *     https://web.ist.utl.pt/~ist11038/compute/errtheory/,regression/regrthroughorigin.pdf
+ */
+
+function fullLinearRegression(X, Y, compute_intercept = true, verbose: boolean = false) {
+  const y = Y.flat();
+  const n = y.length;
+  let { _X } = prepareDataForRegression(X, Y, []);
+  _X = _X.length === n ? transpose2dArray(_X) : _X.slice();
+  assertSameNumberOfElements(_X[0], y);
+  const n_var = _X.length;
+  const n_deg = n - n_var - (compute_intercept ? 1 : 0);
+  /*if (n_var == 1) {
```

Comment:
Should we keep the code for the "simple" case while the "more general" case can also handle it ?